contrib/lang/c-c++/packages.el

```diff
@@ -22,19 +23,34 @@ which require an initialization must be listed explicitly in the list.")
 (defun c-c++/init-cc-mode ()
   (use-package cc-mode
     :defer t
-    :config
+    :init
     (progn
+      (semantic-mode 1)
+
+      (add-to-list 'semantic-default-submodes 'global-semantic-stickyfunc-mode)
+      (add-to-list 'semantic-default-submodes 'global-semantic-idle-summary-mode)
+
       (add-hook 'c-mode-hook '(lambda () (c-toggle-auto-state t)))
       (add-hook 'c++-mode-hook '(lambda () (c-toggle-auto-state t))))))
 
+(defun c-c++/init-srefactor ()
+  (when (not (version< emacs-version "24.4"))
+    (use-package srefactor
+      :init
```

Comment:
in `use-package` there is a keyword `:if` to conditionally enable a package, you can move the tests of emacs version there:

```elisp
:if (version< 24.3 emacs-version)
```